### DISCUSSION TITLE
关于课程2内容的提问

### DISCUSSION THREAD

**Discussion by MorningWeicheng** [mlc-ai/mlc-zh#12] [Q&A] [answered]
请教一下各位，课程2中提到的如果index做了max()操作之后对原先的2轴做reorder()了，针对这一块我不能很直观的理解，不知道可否详细解释下，谢谢！

**Comment by Hzfengsy** [comment, marked as answer]
能不能详细说明一下你的问题，举一个样例代码。或者如果是视频中提到的，麻烦告诉一下时间戳，谢谢

**Reply by tqchen** [reply]
感谢指出这个地方的问题。这个地方是我们的一个错误，实际的例子应该为


```python
for xo in range(32):
  for xi in range(4):
     C[xo * 4 + xi] = A[xo * 4 + xi] + C[max(xo * 4 + xi – 1, 0)] 
```

然后因为在xo = 4, xi=0的情况会对 xo=0, xi=1 的结果造成依赖， 而如果我们改变了循环顺序之后 xo=0,xi=1会先被循环到

**Reply by MorningWeicheng** [reply]
> 感谢指出这个地方的问题。这个地方是我们的一个错误，实际的例子应该为
> 
> ```python
> for xo in range(32):
>   for xi in range(4):
>      C[xo * 4 + xi] = A[xo * 4 + xi] + C[max(xo * 4 + xi – 1, 0)] 
> ```
> 
> 然后因为在xo = 4, xi=0的情况会对 xo=0, xi=1 的结果造成依赖， 而如果我们改变了循环顺序之后 xo=0,xi=1会先被循环到

感谢解答！

**Comment by hbsun2113** [comment]
没懂这个提问，看到原视频里是`B`，纠正为`C`？请提供下出错的时间戳，谢谢！
![image](https://user-images.githubusercontent.com/27994442/175805858-aab42e62-7cee-4336-a728-7b4e8057b648.png)

**Reply by MorningWeicheng** [reply]
是的，可以简单地理解为B纠正为C：
改成C之后--> C = A + C，即C会对其他index下的C的结果存在依赖，如果只是单纯的reorder，循环的顺序会有变化导致依赖关系被打破，会出现等式左边的C计算时等式右边的C还没有结果的情况。
以上是我理解的。

**Reply by hbsun2113** [reply]
谢谢！所以如果是B就可以进行上述的reorder， 是C就不能reorder，对吗？

**Reply by MorningWeicheng** [reply]
嗯 针对讲义中的这一段循环是这样
